packages/remix-dev/compiler.ts

```diff
@@ -311,16 +311,25 @@ async function buildEverything(
 
 async function createBrowserBuild(
   config: RemixConfig,
-  options: BuildOptions & { incremental?: boolean }
+  options: Required<BuildOptions> & { incremental?: boolean }
 ): Promise<esbuild.BuildResult> {
   // For the browser build, exclude node built-ins that don't have a
   // browser-safe alternative installed in node_modules. Nothing should
   // *actually* be external in the browser build (we want to bundle all deps) so
   // this is really just making sure we don't accidentally have any dependencies
   // on node built-ins in browser bundles.
   let dependencies = Object.keys(await getAppDependencies(config));
-  let externals = nodeBuiltins.filter((mod) => !dependencies.includes(mod));
-  let fakeBuiltins = nodeBuiltins.filter((mod) => dependencies.includes(mod));
+
+  let externals = []
+  let fakeBuiltins = [];
+  for (let i = 0; i < nodeBuiltins.length; i++) {
+    let mod = nodeBuiltins[i];
+    if (dependencies.includes(mod)) {
+      fakeBuiltins.push(mod);
+    } else {
+      externals.push(mod);
+    }
+  }
```

Comment:
Doing this in 1 loop might speed things up
Maybe we could use a "for of" loop here?

```ts
const externals = [];
const fakeBuiltins = [];
for(const mod of nodBuiltins) {  
    if (dependencies.includes(mod)) {
      fakeBuiltins.push(mod);
    } else {
      externals.push(mod);
    }
}
```

Reply:
@cliffordfajardo probably using the traditional `for` would be faster than the `for..of`. I'll change it, thanks for the warning (I didn't even notice) o/

Reply:
Although it might be "faster" (which is neglectable in this case imo, it's much less readable than having the built in `filter` methods, so I would rather revert this again.

Reply:
TBH - I do like the `filter `methods more & think they are more readable
When it comes to JS perf, switching from filter to `for` in terms of performance might be negligible (not saying its not important) its just that  given V8 is always changing internally, I don't often think about things like `for` vs `forEach` when the number of item's we're iterating over is low + what get's transpiled to production code javascript code, usually isn't written in the same way we write JS in the codebase & lets not get started on different JS runtimes lol 😅 , which all have different performance charasteristics

Appreciate the thought @MichaelDeBoey